tests/fixtures/saas/hubspot_fixtures.py

```diff
@@ -85,3 +95,45 @@ def dataset_config_hubspot(
     )
     yield dataset
     dataset.delete(db=db)
+
+
+@pytest.fixture(scope="function")
+def setup_teardown_erasure_hubspot_data(connection_config_hubspot, hubspot_erasure_identity_email) -> Generator:
+    """
+    Gets the current value of the resource and restores it after the test is complete.
+    Used for erasure tests.
+    """
+
+    connector = SaaSConnector(connection_config_hubspot)
+
+    # create contact
+    contacts_request: SaaSRequestParams = SaaSRequestParams(
+        method=HTTPMethod.POST,
+        path=f"/crm/v3/objects/contacts",
+        json_body={
+            "properties": {
+                "company": "test company",
+                "email": hubspot_erasure_identity_email,
+                "firstname": "SomeoneFirstname",
+                "lastname": "SomeoneLastname",
+                "phone": "(123) 123-1234",
+                "website": "someone.net"
+            }
+        },
+    )
+    contacts_response = connector.create_client().send(contacts_request)
+    contacts_body = contacts_response.json()
+    contact_id = contacts_body["id"]
+
+    # no need to subscribe contact, since creating a contact auto-subscribes them
+
+    time.sleep(10)  # Allows contact to be propagated in Hubspot before calling access / erasure requests
+
+    yield contact_id
+
+    # delete contact
+    delete_request: SaaSRequestParams = SaaSRequestParams(
+        method=HTTPMethod.DELETE,
+        path=f"/crm/v3/objects/contacts/{contact_id}",
+    )
+    connector.create_client().send(delete_request)
```

Comment:
Should you verify that this was successful?

Reply:
Just to check my understanding, you create some test data, mask it, and then delete the test data as part of cleanup?

Reply:
Ideally we'd check if we were successful here as @pattisdr suggests. If we don't we run the risk of polluting the test account with data from unit tests. I imagine this might need another sleep / healthcheck style code block however?

Reply:
@pattisdr that's correct, we create a new contact, then delete it at the end. If we try to re-use a contact, hubspot does not allow using the API to subscribed them (part of resetting the data after our testing) after a contact has unsubscribed from a subscription.

@seanpreston good idea, I'll make sure it's deleted. Thanks!